.github/workflows/rollback.yml

```diff
@@ -0,0 +1,307 @@
+          aws-region: us-west-2
+      - name: Check version exists on Release S3
+        run: |
+          S3_KEY="${{ matrix.os }}/${{ matrix.arch }}/${{ github.event.inputs.rollback_version }}/aws-otel-collector.${{ matrix.type }}"
+          aws s3api head-object --bucket $S3_BUCKET --key $S3_KEY > /dev/null
+  ssm-check:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - type: global
+            regions: "eu-north-1 ap-south-1 eu-west-3 us-east-2 eu-west-1 eu-central-1 sa-east-1 us-east-1 \
+                  ap-northeast-2 ap-northeast-3 eu-west-2 ap-northeast-1 us-west-2 us-west-1 ap-southeast-1 \
+                  ap-southeast-2 ca-central-1"
+          - type: hkg
+            regions: "ap-east-1"
+          - type: bah
+            regions: "me-south-1"
+          - type: cpt
+            regions: "af-south-1"
+          - type: mxp
+            regions: "eu-south-1"
+          - type: cn
+            regions: "cn-north-1 cn-northwest-1"
+    steps:
+      - name: Configure AWS Credentials
+        if: matrix.type == 'global'
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.RELEASE_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.RELEASE_SECRET }}
+          aws-region: us-west-2
+      - name: Configure AWS Credentials for HKG
+        if: matrix.type == 'hkg'
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.RELEASE_HKG_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.RELEASE_HKG_SECRET }}
+          aws-region: ap-east-1
+      - name: Configure AWS Credentials for BAH
+        if: matrix.type == 'bah'
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.RELEASE_BAH_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.RELEASE_BAH_SECRET }}
+          aws-region: me-south-1
+      - name: Configure AWS Credentials for CPT
+        if: matrix.type == 'cpt'
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.RELEASE_CPT_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.RELEASE_CPT_SECRET }}
+          aws-region: af-south-1
+      - name: Configure AWS Credentials for MXP
+        if: matrix.type == 'mxp'
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.RELEASE_MXP_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.RELEASE_MXP_SECRET }}
+          aws-region: eu-south-1
+      - name: Configure AWS Credentials for CN
+        if: matrix.type == 'cn'
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.RELEASE_CN_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.RELEASE_CN_SECRET }}
+          aws-region: cn-north-1
+      - name: Check version exists in SSM
```

Comment:
Should this happen as a job before this one and have this job require this step?